src/SASViyaApiClient.ts

```diff
@@ -381,7 +381,11 @@ export class SASViyaApiClient {
         )
         const newFolderName = `${parentFolderPath.split('/').pop()}`
         if (newParentFolderPath === '') {
-          throw new Error('Root folder has to be present on the server.')
+          throw new RootFolderNotFoundError(
```

Comment:
This is now its own error type - `RootFolderNotFoundError`. If required, we can handle it differently on the calling side.